diff --git a/README.md b/README.md
index 6cd52025..a1903fdc 100644
--- a/README.md
+++ b/README.md
@@ -46,7 +46,7 @@ Keep in mind that libudev-zero isn't ideal. Here are some pros/cons:
 * android-tools - requires udev rules for non-root usage
 * NetworkManager - needs investigation
 * libgudev - needs investigation
-* PipeWire - depends on udev internal properties. [patch](https://github.com/illiliti/libudev-zero/issues/26#issuecomment-1848802791)
+* PipeWire - Requires helper rules [script](https://github.com/illiliti/libudev-zero/blob/master/contrib/libudev-zero-rules-helper.sh)
 * ldm - depends on udev internal properties
 * lvm2 - uses deprecated `udev_queue` API
 * cups - needs investigation
diff --git a/udev_enumerate.c b/udev_enumerate.c
index 6fd49c09..3d1348e5 100644
--- a/udev_enumerate.c
+++ b/udev_enumerate.c
@@ -278,20 +278,34 @@ static int scan_devices(struct udev_enumerate *udev_enumerate, const char *path)
 
 int udev_enumerate_scan_devices(struct udev_enumerate *udev_enumerate)
 {
-    const char *path[] = { "/sys/dev/block", "/sys/dev/char", NULL };
-    int i;
+    int i, cnt, ret;
+    struct dirent **de;
+
+    ret = 0;
 
     if (!udev_enumerate) {
         return -1;
     }
 
-    for (i = 0; path[i]; i++) {
-        if (!scan_devices(udev_enumerate, path[i])) {
-            return -1;
+    cnt = scandir("/sys/class", &de, filter_dot, NULL);
+
+    if (cnt == -1)
+        return -1;
+
+    for (i = 0; i < cnt; i++) {
+        char device_path[PATH_MAX];
+        snprintf(device_path, sizeof(device_path), "/sys/class/%s", de[i]->d_name);
+        if (!scan_devices(udev_enumerate, device_path)) {
+            ret = -1;
         }
     }
 
-    return 0;
+    for (i = 0; i < cnt; i++) {
+        free(de[i]);
+    }
+
+    free(de);
+    return ret;
 }
 
 /* XXX NOT IMPLEMENTED */ int udev_enumerate_scan_subsystems(struct udev_enumerate *udev_enumerate)
