--- a/cmake/compiler-flags.cmake	2026-06-11 17:11:25.587916900 +0200
+++ b/cmake/compiler-flags.cmake2	2026-06-11 17:22:59.840620962 +0200
@@ -256,7 +256,7 @@
   # -fcf-protection is supported only on x86 GNU/Linux per this gcc doc:
   # https://gcc.gnu.org/onlinedocs/gcc/Instrumentation-Options.html#index-fcf-protection
   if ((CMAKE_SYSTEM_NAME STREQUAL "Linux") AND (CMAKE_SYSTEM_PROCESSOR MATCHES "(x86)|(X86)"))
-    add_compiler_flags(8.1 "-fcf-protection")
+  #  add_compiler_flags(8.1 "-fcf-protection")
   endif ()
   add_compiler_flags(4.6 "-pipe")
   add_compiler_flags(4.6 "-Werror=format-security")
